river/PointerMapping.zig

```diff
@@ -14,13 +14,47 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+const Self = @This();
+
 const wlr = @import("wlroots");
 
+const util = @import("util.zig");
+
 pub const Action = enum {
     move,
     resize,
+    command,
 };
 
 event_code: u32,
 modifiers: wlr.Keyboard.ModifierMask,
 action: Action,
+command_args: []const [:0]const u8,
```

Comment:
You could turn `action` into a tagged union (using `Action` as tag-type). Then `command_args` could be the `command` case, with the two others being `void`.

Reply:
That's a nice construct. Done.